### PR TITLE
🔧 bsx seo preview & canonical links

### LINKED ISSUE #3243: SEO: define canonical URL
Capture d’écran 2022-06-22 à 3 29 55 PM

### LINKED ISSUE #3258: Seems preview cards doesn't works everytime
Seems SEO cards don't work, every time? 
Discord picture 
image

edit on profile works 

image

### PR DESCRIPTION
**Pull request by roiLeo**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [x] Feature

### What's new?

- [x] PR closes #3243 
- [x] PR closes #3258 
- [x] fix bsx GalleryItem seo preview
- [x] add canonical url on item & collection

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=DVYy1qnocE8t6ZvUfPx3rEjG829khNRXx3YrCGVHHj19Lcb)

### Screenshot

Screenshot 2022-06-27 at 11-27-45 Card Validator Twitter Developers

### CODE REVIEW

**Comment by yangwao** [comment]
curious if that would work on `kodadot.xyz` domain as on `beta.kodadot.xyz` it used to work here and there

**Comment by roiLeo** [comment]
> curious if that would work on `kodadot.xyz` domain as on `beta.kodadot.xyz` it used to work here and there

might be related with #2759

**Review by prachi00 — APPROVED** [review]

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 50.12 USD/KSM ~ 1.995 $KSM
🧗 [DVYy1qnocE8t6ZvUfPx3rEjG829khNRXx3YrCGVHHj19Lcb](https://kusama.subscan.io/account/DVYy1qnocE8t6ZvUfPx3rEjG829khNRXx3YrCGVHHj19Lcb)
🔗 [0x43306577632da3ad0c79ff075b4244f1289840dee4b00a71cdc40a7f6bcb065b](https://kusama.subscan.io/extrinsic/0x43306577632da3ad0c79ff075b4244f1289840dee4b00a71cdc40a7f6bcb065b)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by roiLeo** [comment]
> oh wait seems, can't test because have 524 and 429 errors :(
> 
> image

?
I doubt we use cloudflare for seo

Tested with facebook opengraph debugger
Capture d’écran 2022-07-05 à 9 04 55 AM

**Comment by yangwao** [comment]
wfm 

image

**Comment by yangwao** [comment]
> wait for this one, let me remove decimals from BSX amount

ooook

**Comment by yangwao** [comment]
chill we are far beyond worst

**Comment by roiLeo** [comment]
not working as excepted for now will change it a bit latter

**Comment by roiLeo** [comment]
Blocked by #3374 can't make my local env to work (it show rmrk metadata in bsx item)

**Comment by roiLeo** [comment]
Better now without empty decimals an added space on BSX (ref #3038)
Capture

I sense we need to do something to extend polkadot js `formatAmount` method

File: utils/format/balance.ts [hunk 1]
```diff
@@ -10,3 +10,25 @@ export const formatBalanceEmptyOnZero = (
     ? ''
     : trimAll(formatBalance(amount, decimals || 12, symbol || 'KSM'))
 }
+
+export const formatBsxBalanceEmptyOnZero = (
+  amount: string,
+  decimals?: number,
+  symbol?: string
+) => {
```
Comment (kkukelka) [inline utils/format/balance.ts:18]: adding return type here?
```suggestion
): string => {
```

File: components/bsx/Gallery/Item/GalleryItem.vue [hunk 2]
```diff
@@ -208,6 +225,19 @@ export default class GalleryItem extends mixins(
     return [this.collectionId, this.id]
   }
 
+  get pageTitle(): string {
+    return `${this.nft.name || ''}`
```
Comment (kkukelka) [inline components/bsx/Gallery/Item/GalleryItem.vue:229]: no template string needed here
```suggestion
    return this.nft?.name || ''
```

**Review by kkukelka — CHANGES_REQUESTED** [review]

File: components/bsx/Gallery/Item/GalleryItem.vue (hunk 2 above)
Comment (roiLeo) [reply components/bsx/Gallery/Item/GalleryItem.vue:229]: oh no why did I copy this

**Review by kkukelka — APPROVED** [review]
excellent 👍

**Comment by kkukelka** [comment]
pay 150

**Comment by roiLeo** [comment]
myman

**Pull request merged**